packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,257 @@
+  }
+
+  function _validateMintReserve(MintReservation calldata mintReservation) private {
+    if (block.timestamp > mintReservation.expiration) revert ExpiredSignature();
+    // Verify valid message based on input variables
+    bytes32 expectedMessage = keccak256(
+      abi.encodePacked(
+        abi.encode(mintReservation.mints, mintReservation.gid, mintReservation.expiration, mintReservation.nonce, msg.value)
+      )
+    );
+        // address signer = mintReservation.message.recover(mintReservation.signature);
+    if (mintReservation.message != expectedMessage || msg.sender != _signer) revert InvalidSignature();
+    if (_usedNonces[mintReservation.gid][mintReservation.nonce]) revert InvalidNonce();
+    _usedNonces[mintReservation.gid][mintReservation.nonce] = true;
+  }
+
+  function _mint(Mint[] calldata mints) private {
+    for (uint256 i; i < mints.length; ) {
+      Mint calldata mintData = mints[i];
+      _mintClaim(mintData);
+      if (_deliveredMintsPerWallet[mintData.creatorContractAddress][mintData.instanceId][mintData.recipients[i].receiver] + 1 > _reservedMintsPerWallet[mintData.creatorContractAddress][mintData.instanceId][mintData.recipients[i].receiver])
+        revert IGachaLazyClaim.CannotMintMoreThanReserved();
+      _deliveredMintsPerWallet[mintData.creatorContractAddress][mintData.instanceId][mintData.recipients[i].receiver] ++;
+      emit GachaClaimMintDelivered(mintData.creatorContractAddress, mintData.instanceId, mintData.recipients[i].receiver);
+      unchecked {
+        ++i;
+      }
+    }
+  }
+
+  /**
+   * See {IGachaLazyClaim-deliver}.
+   */
+  function deliverMints(IGachaLazyClaim.Mint[] calldata mints) external override {
+    _validateSigner();
+    _mint(mints);
+  }
+
+  /**
+   * Mint a claim
+   */
+  function _mintClaim(Mint calldata mintData) private {
+    uint256[] memory tokenIds = new uint256[](1);
+    tokenIds[0] = mintData.tokenId;
+    address[] memory receivers = new address[](mintData.recipients.length);
+    uint256[] memory amounts = new uint256[](mintData.recipients.length);
+    for (uint256 i; i < mintData.recipients.length; ) {
+      receivers[i] = mintData.recipients[i].receiver;
+      amounts[i] = mintData.recipients[i].mintCount;
+      unchecked {
+        ++i;
+      }
+    }
+    IERC1155CreatorCore(mintData.creatorContractAddress).mintExtensionExisting(receivers, tokenIds, amounts);
+  }
+
+  // airdrop?
+
+  /**
+   * See {IGachaLazyClaim-getUserMints}.
+   */
+  function getUserMints(
+    address minter,
+    address creatorContractAddress,
+    uint256 instanceId
+  ) external view override returns (UserMint memory) {
+    return _getUserMints(minter, creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {ICreatorExtensionTokenURI-tokenURI}.
+   */
+  function tokenURI(address creatorContractAddress, uint256 tokenId) external view override returns (string memory uri) {
+    uint224 tokenClaim = uint224(_claimTokenIds[creatorContractAddress][tokenId]);
```

Comment:
rename 'tokenClaim' to instanceId because that's what it is.  No need to cast here.